# Puzzle

https://thefiddler.substack.com/p/can-you-box-the-letters

# Solution

I am going to take a coding approach, but I'll do some estimation for the fiddler case beforehand to sanity check the results.

If there was no constraint, then the answer would trivially be 8!.

The fiddler answer should be less than 8!, because we put some constraints excluding some patterns.

At each step, we have k choices left, but 0 or 1 may be excluded by being on the same line as the current choice. If we reduce each term in 8! by 1, then we get 7!.

So, the fiddler answer should be between 5040 and 40320.

And my guess for it is

8 * 6 * 5 * 4.5 * 3.5 * 2.5 * 1.5 * 0.5 = 7087

where the 4.5 indicates that the choice can be 4 or 5, so loose average would be 4.5.


In [21]:
from functools import cache
@cache
def count_letterboxed_sequences(choices_0, choices_1, choices_2, choices_3, last_side=-1):
    if choices_0 == 0 and choices_1 == 0 and choices_2 == 0 and choices_3 == 0:
        return 1
    num_sequences = 0
    if choices_0 > 0 and last_side != 0:
        num_sequences += choices_0 * count_letterboxed_sequences(choices_0 - 1, choices_1, choices_2, choices_3, 0)
    if choices_1 > 0 and last_side != 1:
        num_sequences += choices_1 * count_letterboxed_sequences(choices_0, choices_1 - 1, choices_2, choices_3, 1)
    if choices_2 > 0 and last_side != 2:
        num_sequences += choices_2 * count_letterboxed_sequences(choices_0, choices_1, choices_2 - 1, choices_3, 2)
    if choices_3 > 0 and last_side != 3:
        num_sequences += choices_3 * count_letterboxed_sequences(choices_0, choices_1, choices_2, choices_3 - 1, 3)
    #print(f"choices: {choices_0}, {choices_1}, {choices_2}, {choices_3}, {last_side=} => {num_sequences}")
    return num_sequences

assert count_letterboxed_sequences(0, 0, 2, 2) == 8

In [22]:
Fiddler_answer = count_letterboxed_sequences(2,2,2,2)
print(f"The Fiddler answer is: {Fiddler_answer}")

The Fiddler answer is: 13824


In [23]:
Extra_credit_answer = count_letterboxed_sequences(3,3,3,3)
print(f"The Extra Credit answer is: {Extra_credit_answer}")

The Extra Credit answer is: 53529984


In [29]:
from math import factorial
# Just for kicks.
for i in range(1, 61):
    num_sequences = count_letterboxed_sequences(i, i, i, i)
    ratio = num_sequences / factorial(i * 4)
    print(f"{i} letters per side: ratio: {ratio}")

1 letters per side: ratio: 1.0
2 letters per side: ratio: 0.34285714285714286
3 letters per side: ratio: 0.11175324675324676
4 letters per side: ratio: 0.035916845059702204
5 letters per side: ratio: 0.011474857394804321
6 letters per side: ratio: 0.0036547135016853673
7 letters per side: ratio: 0.0011619089793505732
8 letters per side: ratio: 0.0003689687462101741
9 letters per side: ratio: 0.00011707605167168686
10 letters per side: ratio: 3.7128430387165876e-05
11 letters per side: ratio: 1.1769797803190356e-05
12 letters per side: ratio: 3.729909887902065e-06
13 letters per side: ratio: 1.1817474309603094e-06
14 letters per side: ratio: 3.7434291937402635e-07
15 letters per side: ratio: 1.1856300041559678e-07
16 letters per side: ratio: 3.7547012740866436e-08
17 letters per side: ratio: 1.1889333479336572e-08
18 letters per side: ratio: 3.764460723428004e-09
19 letters per side: ratio: 1.1918371214016043e-09
20 letters per side: ratio: 3.7731539385853047e-10
21 letters per side: ra

# Conclusions

The Fiddler answer is: 13824

The Extra Credit answer is: 53529984

The sequence as we explore larger numbers of letters per side has a interesting structure where odd and even entries converge differently.

Even N letters per side entries seem to be ~ (4N)! * 3.88 * 10^(-N/2)

Odd N letters per side entries seem to be ~ (4N)! * 1.22 * 10^(-(N-1)/2)
